### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by ahangarha**
I have messed up a little bit!!

Commit `9387e69` shouldn't be there! I couldn't remove it. so I reverted it at the end.

Another problem is that my vim added tab while space was used for indention in the original code. I could make one extra commit to solve the problem but I was not sure if it was a good practice or not.

Tell me if I can do anything. Feel free to drop the PR

### CODE REVIEW

**Comment by MCSH** [comment]
I'll review the changes, but I highly suggest converting the tabs to spaces, and 2 space indentions is what I have used so far. you can create another commit and then squash all of them together (don't worry if you don't know how, I can do it here as well)

File: models.js [hunk 1]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
   }
 };
 
+class User extends BaseModel{
```
Comment (MCSH) [inline models.js]: I don't think using users is appropriate, we'll be using the actor class activityinn provides.

File: models.js [hunk 2]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		super();
+
+		this.addField('firstName', inn.STRING(64));
+		this.addField('lastName', inn.STRING(64));
+		this.addField('email', inn.STRING(64));
+		this.addField('password', inn.STRING(128));
+	}
+}
+
+class Roles extends BaseModel{
```
Comment (MCSH) [inline models.js]: Again, leave the permission management to the activity inn

File: models.js [hunk 3]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		this.addField('text', inn.STRING(1024));
+	}
+}
+
+
+class Categories extends BaseModel{
+	constructor(){
+		super();
+
+		this.addField('type', inn.STRING(64));
```
Comment (MCSH) [inline models.js]: what's a type for a category? is it same thing as title or name?

File: models.js [hunk 4]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		this.addField('userId', inn.relation.foreignkey(inn.User));
+		this.addField('roleId', inn.relation.foreignkey(inn.Roles));
+	}
+}
+
+class Comments extends BaseModel{
+	constructor(){
+		super();
+
+		this.addField('userId', inn.relation.foreignkey(inn.User));
```
Comment (MCSH) [inline models.js]: I'd change the inn.User to Inn.Actor

File: models.js [hunk 5]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		this.addField('postId', inn.relation.foreignkey(inn.Post));
```
Comment (MCSH) [inline models.js]: Inn does not have a Post object, we have defined it so you should just say Post, without the inn at the beginning.

File: models.js [hunk 6]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		this.addField('type', inn.STRING(64));
+		this.addField('desctiption', inn.STRING(128));
+	}
+}
+
+class CategoriesPost extends BaseModel{
+	constructor(){
+		super();
+
+		this.addField('categoryId', inn.relation.foreignkey(inn.Categories));
```
Comment (MCSH) [inline models.js]: inn.Categories => Categories

File: models.js [hunk 7]
```diff
@@ -30,7 +30,81 @@ class Post extends BaseModel{
...
+		super();
+
+		this.addField('userId', inn.relation.foreignkey(inn.User));
+		this.addField('postId', inn.relation.foreignkey(inn.Post));
+		this.addField('text', inn.STRING(1024));
+	}
+}
+
+
+class Categories extends BaseModel{
```
Comment (MCSH) [inline models.js]: Shouldn't the class name be Category? i.e: Single, not plural? Same thing for other classes.

File: models.js (hunk 4 above)
Comment (MCSH) [reply models.js]: Also inn.relation should be Inn.relation, same for every other field in this file.

File: models.js (hunk 7 above)
Comment (ahangarha) [reply models.js]: sure. will be changed

File: models.js (hunk 2 above)
Comment (ahangarha) [reply models.js]: will be removed

File: models.js (hunk 3 above)
Comment (ahangarha) [reply models.js]: I think this is related to the type of taxonomy. in wordpress, by default there are two taxonomies: category, tag. Here I guess the idea is the same. Yet, we can remove this for simplicity.

File: models.js (hunk 4 above)
Comment (ahangarha) [reply models.js]: Now, we don't have inn or Inn. we have models. should I replace userId with actor and set it to `models.relation.foreignkey(models.Actor)`?

File: models.js (hunk 3 above)
Comment (MCSH) [reply models.js]: Sure, your call to leave it or remove it.

File: models.js (hunk 4 above)
Comment (MCSH) [reply models.js]: I'd go with `models.relation.foreignkey(models.Actor)`, I think that's the better option here.

**Comment by ahangarha** [comment]
I think I have made the required changes. Make a fresh review please

**Comment by MCSH** [comment]
Looks good!

Only we forgot to register the models in `index.js`, something like this for other models:
```
const {Blog, Post} = require('./models');
inn.register(Blog);
inn.register(Post);
```

**Comment by ahangarha** [comment]
I have committed the required changes for registration of models in index.js. I also fixed a typo there.

**Comment by MCSH** [comment]
Look great, thanks!

**Pull request merged**